test/performance/range/views/view_translate_benchmark.cpp

```diff
@@ -0,0 +1,306 @@
+
+template <typename container_t, typename adaptor_t>
+void copy_to_vector(benchmark::State & state)
+{
+    container_t c(2000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+    {
+        c[i] = generate_sequence<seqan3::dna4>(5000, 0, 0);
+    }
+
+    auto adaptor = views::deep{adaptor_t{}};
+
+    for (auto _ : state)
+    {
+        container_t s{c};
+        std::vector<std::vector<std::vector<aa27>>> vec = s | adaptor
+                                                            | views::to<std::vector<std::vector<std::vector<aa27>>>>;
+    }
+}
+
+BENCHMARK_TEMPLATE(copy_to_vector, std::vector<std::vector<dna4> >, seqan3::detail::translate_fn<false>);
+
+// ============================================================================
+//  copy_to_vector translate_join
+// ============================================================================
+
+template <typename container_t, typename adaptor_t>
+void copy_to_vector_join(benchmark::State & state)
+{
+    container_t c(2000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+    {
+        c[i] = generate_sequence<seqan3::dna4>(5000, 0, 0);
+    }
+
+    auto adaptor = adaptor_t{};
+
+    for (auto _ : state)
+    {
+        container_t s{c};
+        auto v = s | adaptor;
+        std::vector<std::vector<aa27>> vec = s | adaptor
+                                               | views::to<std::vector<std::vector<aa27>>>;
+    }
+}
+
+BENCHMARK_TEMPLATE(copy_to_vector_join, std::vector<std::vector<dna4> >, seqan3::detail::translate_join_fn);
+
+// ============================================================================
+//  copy_to_vector SeqAn2
+// ============================================================================
+
+#ifdef SEQAN3_HAS_SEQAN2
+
+template <typename container_t>
+void seqan2_copy_to_vector(benchmark::State & state)
+{
+    container_t c;
+    resize(c, 2000);
+
+    for (size_t i = 0; i < length(c); ++i)
+    {
+        c[i] = generate_sequence_seqan2<seqan::Dna>(5000, 0, 0);
+    }
+
+    seqan::StringSet<seqan::String<seqan::AminoAcid>, seqan::Owner<seqan::ConcatDirect<> > > out;
+
+    for (auto _ : state)
+    {
+        container_t s{c};
+        seqan::translate(out, s, seqan::SIX_FRAME, seqan::CANONICAL, seqan::Serial());
+    }
+}
+
+BENCHMARK_TEMPLATE(seqan2_copy_to_vector, seqan::StringSet<seqan::DnaString>);
+
+template <typename container_t>
+void seqan2_copy_to_vector_parallel(benchmark::State & state)
+{
+    container_t c;
+    resize(c, 2000);
+
+    for (size_t i = 0; i < length(c); ++i)
+    {
+        c[i] = generate_sequence_seqan2<seqan::Dna>(5000, 0, 0);
+    }
+
+    seqan::StringSet<seqan::String<seqan::AminoAcid>, seqan::Owner<seqan::ConcatDirect<> > > out;
+
+    for (auto _ : state)
+    {
+        container_t s{c};
+        seqan::translate(out, s, seqan::SIX_FRAME, seqan::CANONICAL, seqan::Parallel());
+    }
+}
+
+BENCHMARK_TEMPLATE(seqan2_copy_to_vector_parallel, seqan::StringSet<seqan::DnaString>);
```

Comment:
The SeqAn2 benchmarks can be merged into one with `seqan::Parallel` / `seqan::Serial` being another template parameter.